wcfsetup/install/files/lib/data/smiley/Smiley.class.php

```diff
@@ -75,26 +76,56 @@ public function getAliases()
     }
 
     /**
-     * Returns the height of the smiley.
+     * Returns the dimensions of the smiley.
      *
-     * @return  int
+     * @return  array
      */
-    public function getHeight()
+    public function getDimensions()
     {
-        if ($this->height === null) {
-            $this->height = 0;
+        if ($this->height === null && $this->width === null) {
+            $this->height = $this->width = 0;
 
             $file = WCF_DIR . $this->smileyPath;
             if (\file_exists($file) && \preg_match('~\.(gif|jpe?g|png)$~', $file)) {
                 $data = \getimagesize($file);
                 if ($data !== false) {
+                    // index '0' contains the width of the image
+                    $this->width = $data[0];
+
                     // index '1' contains the height of the image
                     $this->height = $data[1];
```

Comment:
```suggestion
                    [$this->width, $this->height] = $data;
```

see: https://3v4l.org/kMjLa